app.py

```diff
@@ -32,6 +31,25 @@ def wrapper(*args, **kw):
     return wrapper
 
 
+def recover_identity(f):
+    @wraps(f)
+    def wrapper(*args, **kw):
+        authorization = request.headers.get('Authorization')
+        authentication_type, signature_base64_encoded = authorization.split(' ')
+        if authentication_type != 'Signature':
```

Comment:
Validate `signature_base64_encoded` to be not empty and dont start crypto procedures

Reply:
added validation